SepoliaMetamark/web3/contracts/CrowdFunding.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+contract CrowdFunding {
+    struct Campaign {
+        address owner;
+        string title;
+        string description;
+        uint256 target;
+        uint256 deadline;
+        uint256 amountCollected;
+        string image;
+        address[] donators;
+        uint256[] donations;
+    }
+
+    mapping(uint256 => Campaign) public campaigns;
+
+    uint256 public numberOfCampaigns = 0;
+
+    function createCampaign(address _owner, string memory _title, string memory _description, uint256 _target, uint256 _deadline, string memory _image) public returns (uint256) {
+        Campaign storage campaign = campaigns[numberOfCampaigns];
+
+        require(campaign.deadline < block.timestamp, "The deadline should be a date in the future.");
+
+        campaign.owner = _owner;
+        campaign.title = _title;
+        campaign.description = _description;
+        campaign.target = _target;
+        campaign.deadline = _deadline;
+        campaign.amountCollected = 0;
+        campaign.image = _image;
+
+        numberOfCampaigns++;
+
+        return numberOfCampaigns - 1;
+    }
+
+    function donateToCampaign(uint256 _id) public payable {
+        uint256 amount = msg.value;
+
+        Campaign storage campaign = campaigns[_id];
+
+        campaign.donators.push(msg.sender);
+        campaign.donations.push(amount);
+
+        (bool sent,) = payable(campaign.owner).call{value: amount}("");
```

Comment:
what is the purpose of this `payable` function? How can we withdraw the fund?

Reply:
> what is the purpose of this `payable` function? How can we withdraw the fund?

When donating, the money will be transferred to the campaign creator's metamark wallet 🤔